tests/liteclientHSM_test.cc

```diff
@@ -0,0 +1,1179 @@
+  OSTreeRepoMock repo_;
+};
+
+/**
+ * Class TufRepoMock
+ *
+ */
+class TufRepoMock {
+ public:
+  TufRepoMock(const boost::filesystem::path& _root, std::string expires = "",
+              std::string correlation_id = "corellatio-id")
+      : root_{_root.string()}, repo_{_root, expires, correlation_id}, latest_{Uptane::Target::Unknown()} {
+    repo_.generateRepo(KeyType::kED25519);
+  }
+
+ public:
+  const std::string& getPath() const { return root_; }
+  const Uptane::Target& getLatest() const { return latest_; }
+
+  Uptane::Target addTarget(const std::string& name, const std::string& hash, const std::string& hardware_id,
+                           const std::string& version, const Json::Value& apps_json = Json::Value()) {
+    Delegation null_delegation{};
+    Hash hash_obj{Hash::Type::kSha256, hash};
+
+    Json::Value custom_json;
+    custom_json["targetFormat"] = "OSTREE";
+    custom_json["version"] = version;
+    custom_json[Target::ComposeAppField] = apps_json;
+    repo_.addCustomImage(name, hash_obj, 0, hardware_id, "", null_delegation, custom_json);
+
+    Json::Value target_json;
+    target_json["length"] = 0;
+    target_json["hashes"]["sha256"] = hash;
+    target_json["custom"] = custom_json;
+    latest_ = Uptane::Target(name, target_json);
+    return latest_;
+  }
+
+ private:
+  const std::string root_;
+  ImageRepo repo_;
+  Uptane::Target latest_;
+};
+
+/**
+ * Class DeviceGatewayMock
+ *
+ */
+class DeviceGatewayMock {
+ public:
+  static std::string RunCmd;
+
+ public:
+  DeviceGatewayMock(const OSTreeRepoMock& ostree, const TufRepoMock& tuf, std::string certDir)
+      : ostree_{ostree},
+        tuf_{tuf},
+        port_{TestUtils::getFreePort()},
+        url_{"https://localhost:" + port_},
+        req_headers_file_{tuf_.getPath() + "/headers.json"},
+        process_{
+            RunCmd,           "--port",          port_,    "--ostree", ostree_.getPath(), "--tuf-repo", tuf_.getPath(),
+            "--headers-file", req_headers_file_, "--mtls", certDir} {
+    sleep(1);
+    LOG_INFO << "Device Gateway is running on port " << port_;
+  }
+
+  ~DeviceGatewayMock() {
+    process_.terminate();
+    process_.wait_for(std::chrono::seconds(10));
+  }
+
+ public:
+  std::string getTreeUri() const { return url_; }
+  std::string getOsTreeUri() const { return url_ + "/treehub"; }
+  std::string getTufRepoUri() const { return url_ + "/repo"; }
+  const std::string& getPort() const { return port_; }
+  Json::Value getReqHeaders() const { return Utils::parseJSONFile(req_headers_file_); }
+
+ private:
+  const OSTreeRepoMock& ostree_;
+  const TufRepoMock& tuf_;
+  const std::string port_;
+  const std::string url_;
+  const std::string req_headers_file_;
+  boost::process::child process_;
+};
+
+std::string DeviceGatewayMock::RunCmd;
+
+class RootCaPKI {
+ public:
+  RootCaPKI(std::string path, std::string key, std::string crt)
+      : key_(path + std::move(key)), crt_(path + std::move(crt)) {
```

Comment:
Looks like there is no much sense in passing `path` into the ctor as a client can pass full paths to `key` and `crt` or as an option, just a folder can be passed into ctor and assumed key and cert names (ca.key and ca.crt)?